### ISSUE TITLE
Enable Recursive Search/Filtering for HTTP Serve Mode

### ISSUE THREAD

**Issue by dmrbatuhan** [rclone/rclone#8085]
The associated forum post URL from `https://forum.rclone.org/t/filter-recursively-while-serving-as-http/47858/3`

What is your current rclone version (output from `rclone version`)?
 rclone v1.68.0                                                                  
 - os/version: ubuntu 22.04 (64 bit)                                             
 - os/kernel: 5.15.0-119-generic (x86_64)                                        
 - os/type: linux                                                                
 - os/arch: amd64                                                                
 - go/version: go1.23.1                                                          
 - go/linking: static                                                            
 - go/tags: none 

#### What problem are you are trying to solve?
Currently, when using rclone serve http to serve files from a remote, the search/filter functionality only works within the currently visible directory. If a file is in a subdirectory, it does not appear in the filter results when searching from the top-level directory. For example, if there is a folder named test that contains xx.mp3, filtering with the keyword x from the top-level directory won't return the file until navigating into the test folder.

How do you think rclone should be changed to solve that?
I would like rclone's http serve mode to support recursive search/filter functionality across all subdirectories when filtering from the top-level directory. This way, users can search for files (e.g., xx.mp3) across all directories without having to manually navigate into each one.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
Can you write down what rclone commands you tried please?

I think --include "*.mp3" should recursively include just mp3 files

**Comment by dmrbatuhan** [comment]
type rclone serve http remote: and go that url, and type something. you will se it won't search til you click the required folder.

**Comment by ncw** [comment]
I see what you mean now, thanks.

That search is just done by JavaScript on the page, it never goes back to the server.

We would need to recursively scan the the directory and all sub directories to find stuff.

It's not impossible...

**Comment by dmrbatuhan** [comment]
Yeah, it would be really handy. Otherwise i really have to mount it to public place of any web server, and code a backend to make some filters, etc. Like, I had been looking for solution to list my files on http or https and i saw the rclone serve thing, but it has this issue.
